### PR TITLE
Update Guide-Headers.md

### PR DESCRIPTION
**Pull request by vonovak**
Adding some information to the section on 'Header interaction with screen component'. In my app which has ~30kLOC I have only *one* `setParams` call because I use mobx for state management, meaning I instantiate an object with all the observable data and functions a component needs, and then navigate to that screen, passing the object through `navigation.state`.

Then, both `static navigationOptions` block and the component render() method have access to the object, and I do not need to use `setParams`. This is also a benefit for testing.

I feel like this should be mentioned in the docs, and I want to ask some redux user (since I mentioned redux in the text) if what I wrote makes sense to them - if it is possible to work with redux like this (I use mobx so I don't know). Thanks!

### CODE REVIEW

**Comment by kelset** [comment]
LGTM, maybe add a link to the libraries? 
https://github.com/mobxjs/mobx 
https://github.com/reactjs/redux

File: docs/guides/Guide-Headers.md [hunk 1]
```diff
@@ -136,3 +136,5 @@ class EditInfoScreen extends React.Component {
 
 
 To see the rest of the header options, see the [navigation options document](/docs/navigators/navigation-options#Stack-Navigation-Options).
+
+As an alternative to `setParams`, you may want to consider using a state management library such as MobX or Redux, and when navigating to a screen, pass an object which contains the data necessary for the screen to render, as well as functions you may want to call that modify the data, make network requests and etc. That way, both your screen component and the static `navbarOptions` block will have access to the object. When following this approach, make sure to consider deep linking, which works best in cases where only javascript primitives are passed as navigation props to your screen. In case when deep linking is necessary, you may use a HOC to transform the primitives to the object your screen components expects.
```
Comment (matthamil) [inline docs/guides/Guide-Headers.md]: NIt: Beginners might not be aware of what a HOC is. Can you change "HOC" to "Higher Order Component (HOC)" and include a link to the React docs? https://reactjs.org/docs/higher-order-components.html 😄

**Comment by vonovak** [comment]
I have added links to HOC docs, mobx and redux

**Comment by matthamil** [comment]
Awesome, looks good. Let's ship it 🚀

**Comment by kelset** [comment]
Thanks @vonovak & @matthamil 💪

**Pull request merged**